src/NLog.Targets.AppCenter/AppCenterTarget.cs

```diff
@@ -192,10 +199,65 @@ private void TrackEvent(string eventName, Exception exception, IDictionary<strin
                 properties = properties ?? new Dictionary<string, string>(1);
                 if (properties.Count < 20)
                     properties["EventName"] = eventName;
-                Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties);
+
+                var path = RenderLogEvent(PathToCrashAttachmentDirectory, LogEventInfo.CreateNullEvent());
+                if (!string.IsNullOrEmpty(path))
+                {
+                    var errorAttachmentLogs = GetCompressedErrorAttachmentLogs(path);
+                    Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties, errorAttachmentLogs.ToArray());
+                }
+                else
+                    Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties);
             }
 
             Microsoft.AppCenter.Analytics.Analytics.TrackEvent(eventName, properties);
         }
+
+        /// <remarks>
+        ///     The maximum attached file (compressed) size is 1mb.
+        ///     The maximum attached file count is 10.
+        /// </remarks>
+        private List<Microsoft.AppCenter.Crashes.ErrorAttachmentLog> GetCompressedErrorAttachmentLogs(string path)

```

Comment:
Can you rename this method to `BuildAttachmentLogs` and change input-parameter to be `LogEventInfo logEvent` ?
Then change the method to return an array `Microsoft.AppCenter.Crashes.ErrorAttachmentLog[]` and then pass this array as input to the `TrackEvent`-method (Return Array.Empty when no path or no files in path).

Thus calling `BuildAttachmentLogs(logEvent)` in `void Write(LogEventInfo logEvent)` just like with `BuildProperties(logEvent)`.